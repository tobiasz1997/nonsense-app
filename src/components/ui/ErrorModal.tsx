import { FC } from 'react';
import Button from '@components/ui/Button';
import Modal from '@components/ui/Modal';
import ModalBox from '@components/ui/ModalBox';

type Props = {
	onClose: () => void;
	onAction: () => void;
	label?: string;
};

const ErrorModal: FC<Props> = (props) => {
	return (
		<Modal>
			<ModalBox onClose={props.onClose}>
				<h2 className="text-xl font-bold text-orange md:text-3xl">Error</h2>
				<p className="text-green-dark dark:text-yellow mt-3 mb-5">
					Something went wrong! Try again.
				</p>
				<div className="flex justify-center">
					<Button className="max-w-max" onClick={props.onAction}>
						{props.label ?? 'OK'}
					</Button>
				</div>
			</ModalBox>
		</Modal>
	);
};

export default ErrorModal;
