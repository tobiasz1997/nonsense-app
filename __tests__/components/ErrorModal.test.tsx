import { render, screen } from '@testing-library/react';
import ErrorModal from '@components/ui/ErrorModal';

describe('Error Modal', () => {
	const label = 'Test';
	const action = jest.fn(() => null);

	it('should render without crashing', () => {
		const { container } = render(
			<ErrorModal onAction={action} onClose={action} />
		);
		expect(container).toMatchSnapshot();
	});

	it('should button label be correct', () => {
		render(<ErrorModal onAction={action} onClose={action} label={label} />);

		const labelInDoc = screen.getByText(label);

		expect(labelInDoc).toBeInTheDocument();
	});
});
