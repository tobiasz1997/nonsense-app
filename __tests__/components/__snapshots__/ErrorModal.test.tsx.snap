// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Modal should render without crashing 1`] = `<div />`;
